test/tests.spec.ts

```diff
@@ -669,6 +669,51 @@ describe("node-saml /", function () {
         expect(profile).to.not.have.property("evilcorp.roles");
       });
 
+      it("valid xml document with multiple SubjectConfirmation should validate", async () => {
+        fakeClock = sinon.useFakeTimers(Date.parse("2020-09-25T16:00:00+00:00"));
+        const base64xml = fs.readFileSync(
+          __dirname + "/static/response.root-signed.message-signed-double-subjectconfirmation.xml",
+          "base64"
+        );
+        const container = { SAMLResponse: base64xml };
+        const signingCert = fs.readFileSync(__dirname + "/static/cert.pem", "utf-8");
+        const privateKey = fs.readFileSync(__dirname + "/static/key.pem", "utf-8");
+
+        const samlObj = new SAML({
+          cert: signingCert,
+          privateKey: privateKey,
+          issuer: "onesaml_login",
+          audience: false,
+        });
+
+        const { profile } = await samlObj.validatePostResponseAsync(container);
+        assertRequired(profile, "profile must exist");
+        expect(profile.nameID).to.equal("vincent.vega@evil-corp.com");
+      });
+
+      it("valid xml document with multiple SubjectConfirmation should validate, first is expired so it should take the second one", async () => {
```

Comment:
While I agree with the premise, I don't see any assertions to guarantee that the second one was the one that was actually used.

Reply:
I agree with you but unfortunately I have no way to verify which one was taken into account. At least as far as I can see.

Reply:
I did it for you. You just had to make the second `SubjectConfirmationData` invalid, and then verify that the validation of it fails.